core/vm/operations_acl.go

```diff
@@ -66,7 +69,11 @@ func makeGasSStoreFunc(clearingRefund uint64) gasFunc {
 			}
 			// EIP-2200 original clause:
 			//		return params.SstoreResetGasEIP2200, nil // write existing slot (2.1.2)
-			return cost + (params.SstoreResetGasEIP2200 - params.ColdSloadCostEIP2929), nil // write existing slot (2.1.2)
+			if evm.chainRules.IsGingerbread {
```

Comment:
Every case it seems to be perfect, but I would add a little refactor.
Instead of checking for the gingerBread fork every time we use it, I would add a function
```
func getColdSloadCostEIP2929(evm *EVM) uint64 {
   if evm.chainRules.IsGingerbread {
      return params.ColdSloadCostEIP2929
   }
   return params.CeloColdSloadCostEIP2929
}
```
and then in every place where the `params.ColdSloadCostEIP2929` was called, just put `getColdSloadCostEIP2929(evm)`.
This way, everyone of this function will have less changes from upstream, and our logic of the gasCost will be in only one little function

Reply:
That makes stuff a lot more readable!